ext/http/01_http.js

```diff
@@ -84,6 +84,11 @@
         throw error;
       }
       if (nextRequest == null) {
+        // Work-around for servers (deno_std/http in particular) that call
```

Comment:
I believe this is indicative of a problem with our API, in particular two dubious practices clash here:
* `HttpConn` is automatically closed when `HttpConn.nextRequest()` fails or a return an error.
* `Deno.upgradeWebSocket()` doesn't actually do anything until the caller uses (returns) the response created by `Deno.upgradeWebSocket()`.

Reply:
> Deno.upgradeWebSocket() doesn't actually do anything until the caller uses (returns) the response created by Deno.upgradeWebSocket()

Why is this a problem? This was done intentionally. The `Deno.upgradeWebSocket` call is just a glorified `new Response()`.